src/util.c

```diff
@@ -576,3 +581,24 @@ bool check_is_ip_addr_valid_local(int ss_family, char *ip_to_check)
 	freeifaddrs(ifaddrp);
 	return is_valid;
 }
+
+char clear_current_line(void)
+{
+	static int is_stdout_a_tty = -1;
```

Comment:
Thanks for the code.

Can we make this as a member of tep? We can check is current term a tty or it is redirected to file when initializing so that we don't need to check it every time when print.

Reply:
I suggest to detect the terminal type in new_ntttcp_test_endpoint() in ntttcp.c, and make the detection result as a bool member variable of tep (such as tep->out_tty).

Reply:
This is not checked every time when printing: the result is cached in a `static` variable, which is carefully set to `-1` when initializing, so the check below (`is_stdout_a_tty < 0`) is as cheap as possible.  So the checks are performed only once.

In any case, this isn't called in any performance-critical parts of the code: it's only called twice a second when giving a printout of the current bandwidth measurement.